googlesource/acceptance_test.go

```diff
@@ -0,0 +1,273 @@
+	return d.Config.GoleakOptions
+}
+
+// WriteToSource writes data for source to pull data from
+func (d AcceptanceTestDriver) WriteToSource(t *testing.T, records []sdk.Record) []sdk.Record {
+	var err error
+	is := is.New(t)
+	config := d.SourceConfig(t)
+	records, err = writeToSource(t, config, records)
+	is.NoErr(err)
+
+	return records
+}
+
+func writeToSource(t *testing.T, config map[string]string, records []sdk.Record) (result []sdk.Record, err error) {
+	result, err = dataSetupWithRecord(t, config, records)
+	return result, err
+}
+func (d AcceptanceTestDriver) ReadFromDestination(*testing.T, []sdk.Record) []sdk.Record {
+	return []sdk.Record{}
+}
+
+func (d AcceptanceTestDriver) GenerateRecord(_ *testing.T) sdk.Record {
+	// we don't create record over here. Because we need to code here and then decode again
+	return sdk.Record{}
+}
+
+func createDatasetForAcceptance(t *testing.T, tableID string) (err error) {
+	ctx := context.Background()
+
+	client, err := bigquery.NewClient(ctx, projectID, option.WithCredentialsJSON([]byte(serviceAccount)))
+	if err != nil {
+		return fmt.Errorf("bigquery.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	meta := &bigquery.DatasetMetadata{
+		Location: location, // See https://cloud.google.com/bigquery/docs/locations
+	}
+
+	// create dataset
+	if err := client.Dataset(datasetID).Create(ctx, meta); err != nil && !strings.Contains(err.Error(), "duplicate") {
+		return err
+	}
+	t.Log("Dataset created")
+	client, err = bigquery.NewClient(ctx, projectID, option.WithCredentialsJSON([]byte(serviceAccount)))
+	if err != nil {
+		return fmt.Errorf("bigquery.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	sampleSchema := bigquery.Schema{
+		{Name: "abb", Type: bigquery.StringFieldType},
+		{Name: "name", Type: bigquery.StringFieldType},
+	}
+
+	metaData := &bigquery.TableMetadata{
+		Schema:         sampleSchema,
+		ExpirationTime: time.Now().AddDate(1, 0, 0), // Table will be automatically deleted in 1 year.
+	}
+	tableRef := client.Dataset(datasetID).Table(tableID)
+	err = tableRef.Create(ctx, metaData)
+	if err != nil && !strings.Contains(err.Error(), "duplicate") {
+		return err
+	}
+	return
+}
+
+// dataSetupWithRecord Initial setup required - project with service account.
+func dataSetupWithRecord(t *testing.T, config map[string]string, record []sdk.Record) (result []sdk.Record, err error) {
+	ctx := context.Background()
+	tableID := config[googlebigquery.ConfigTableID]
+
+	client, err := bigquery.NewClient(ctx, projectID, option.WithCredentialsJSON([]byte(serviceAccount)))
+	if err != nil {
+		return result, fmt.Errorf("bigquery.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	var query string
+	positions := ""
+
+	for i := 0; i < len(record); i++ {
+		name := fmt.Sprintf("name%v", time.Now().AddDate(0, 0, globalCounter).Format("20060102150405"))
+		abb := fmt.Sprintf("name%v", time.Now().AddDate(0, 0, globalCounter).Format("20060102150405"))
```

Comment:
Can we use a local counter instead? Global variables is something which needs to be avoided.

Reply:
There  is an acceptance test case where we create data, sync them and then again create new date and sync it. 
We pull only incremental data. If we don't use a global variable here and make sure we are increaing the count, the data created is smaller than already synced data by test case. And so it does not gets synced. 
Test case I am talking about - `TestSource_Open_ResumeAtPositionCDC`

Reply:
each new record is created as - `fmt.Sprintf("name%v", time.Now().AddDate(0, 0, globalCounter)` 
and we keep on increasing the value of globalCounter. This means the we add one day to the last date. 
If we don't keep a global variable then on every function call we start with time.Now()
eg, 
we have 10 records-
from - `name20220601` to `name20220610`
in order to sync any new record we want our new record to be starting from `name20220611` which is greater than `name20220610`
if we don't don't have a global variable we again start getting record as - `name20220601` because time.Now() will have same date. And any record till - `name20220610` will not be synced. This is because will sync using incremental column name.